catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -13,106 +13,167 @@
  */
 package ddf.catalog.cache.impl;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
 import java.util.function.Function;
 
-import ddf.catalog.data.Attribute;
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
 import ddf.catalog.data.AttributeDescriptor;
 import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
 
+/**
+ * Class used to compare two metacards to determine if the resource associated with it has changed
+ * and needs to be re-retrieved.
+ * <p/>
+ * Note: The metacard resource size is not included in the comparison because it cannot
+ * be relied upon, i.e., federated sources may not return it, may return an inaccurate value
+ * or may return a value in a different unit (kilobytes instead of bytes for instance).
+ * <p>
+ * Since the {@code MetacardResourceSizePlugin} is used to address this issue by updating the
+ * size of the metacard resource after a query is run based on the size of the product in the cache,
+ * using the size would cause two problems. First, the resource would be downloaded twice
+ * before the size in the cached metacard and the one retrieved become identical, which is not
+ * acceptable in low bandwidth environments.
+ * <p>
+ * Secondly, once the resource has been retrieved twice and the size had become accurate,
+ * comparing it becomes meaningless since the {@code MetacardResourceSizePlugin} will always make
+ * the size of the freshly retrieved metacard the same as the one in the cache.
+ */
 public class MetacardComparator {
```

Comment:
Made the class package-private and renamed to `CachedResourceMetacardComparator`.